js/react_native/android/src/main/java/ai/onnxruntime/reactnative/OnnxruntimeModule.java

```diff
@@ -103,39 +131,49 @@ public void run(String key, ReadableMap input, ReadableArray output, ReadableMap
    * @return model loading information, such as key, input names, and output names
    */
   public WritableMap loadModel(String uri, ReadableMap options) throws Exception {
-    InputStream modelStream = reactContext.getApplicationContext().getContentResolver().openInputStream(Uri.parse(uri));
-    WritableMap resultMap = loadModel(uri, modelStream, options);
-    modelStream.close();
-    return resultMap;
+    return loadModelImpl(uri, null, options);
   }
 
   /**
-   * Load a model from raw resource directory.
+   * Load a model from buffer.
    *
-   * @param uri uri parameter from react native loadModel() or dummy string for unit testing purpose
-   * @param modelStream a input stream to read a model
+   * @param modelData the model data buffer
    * @param options onnxruntime session options
    * @return model loading information, such as key, input names, and output names
    */
-  public WritableMap loadModel(String uri, InputStream modelStream, ReadableMap options) throws Exception {
-    OrtSession ortSession = null;
-
-    if (!sessionMap.containsKey(uri)) {
-      byte[] modelArray = null;
+  public WritableMap loadModel(byte[] modelData, ReadableMap options) throws Exception {
+    return loadModelImpl("", modelData, options);
+  }
 
+  /**
+   * Load model implementation method for either from model path or model data buffer.
+   *
+   * @param uri uri parameter from react native loadModel()
+   * @param modelData model data buffer
+   * @param options onnxruntime session options
+   * @return model loading information map, such as key, input names, and output names
+   */
+  public WritableMap loadModelImpl(String uri, byte[] modelData, ReadableMap options) throws Exception {
```

Comment:
yes. it can be pulled into a helper function.

but put it this way as for ios support, we might need to call ort.createSession with both modelPath and model data array - two paths, can be separated this way. just want to match the same pattern there.